CONTRIBUTING.md

```diff
@@ -0,0 +1,30 @@
+# Issues
+
+Issues have three types: bug, feature request, and support. When reporting a bug, you must include
+the operating system used, any relevant information about the tech stack, and the feature flags
+used. 
+
+# Pull Requests
+
+Pull requests should be named with the format `crate: short description of change`, and should use
+lower case letters. Always make a pull request from a new branch, that is named similarly, but with
```

Comment:
Should we really enforce branch names? Seems harsh. I think just `Always make a pull request from a feature branch with a related title` is enough and stops the main issue of people working off `trunk` on their forks.

Reply:
Up for discussion, I'm fine with either, but we should start denying PRs from trunk

Reply:
I'd say just go for banning PRs from trunk so its not too strict but stays reasonable. Branch names can be fun to title sometimes :)

Reply:
i don't understand this restriction.

Reply:
If someone PRs trunk->trunk, it is hard to pull in the proposed changes into the same folder and dev environment as one's own fork, making testing harder. One must clone the entire fork into a new directory to test. If new branches are made, and with a relatively unique name, it's easier to add the contributor's fork as a remote and pull in changes that way.

Reply:
You can change the local name with `git checkout -b <branchname> --track upstream/<branchname>`.

Reply:
I didn't know this, thanks. Still want to enforce this rule.